### PR TITLE
networkd: add support for accept ra

### PR DESCRIPTION
**Pull request by ssahani**
This is a WIP patch for #969

### CODE REVIEW

**Comment by poettering** [comment]
Looks pretty cool, but I think it needs to be slightly more complex. The kernel defaults it to on if ip forwarding is off on the iface, ad to off if ip forwarding is on. We should enforce a similar logic.

I figure this means we habe to make this a tristate, i.e. and int, that can take < 0, > 0 and 0, for "unset", "true" and "false". Then, when we about to write the sysctl into the kernel, we should check if the value is >= 0, if so, we configure the value directly. However, it if is < 0 (i.e. unset), we check the ip forwarding setting we maintain for the interface, and then set it to depending on that. That way we mimic the kernel's behaviour.

There's also ready a parse for such tristate variables, in `config_parse_tristate()`. We hence just need to change the type of the field from bool to "int", and then make use of this call.

**Comment by ssahani** [comment]
Yes makes sense. Will update the patch.

**Comment by ssahani** [comment]
Updated  the patch .

File: src/network/networkd-link.c [hunk 1]
```diff
@@ -1789,6 +1799,61 @@ static int link_set_ipv6_privacy_extensions(Link *link) {
...
+         */
+        if (link->network->ipv6_accept_ra == -1) {
+
+                p = strjoina("/proc/sys/net/ipv6/conf/", link->ifname, "/forwarding");
+                r = read_one_line_file(p, &content);
+                if (r >= 0) {
+                        r = safe_atolu(content, &ip_forward);
+                        if (r < 0) {
+                                log_link_warning_errno(link, r,
+                                                       "Invalied ip forwarding setting for the interface: %m");
```
Comment (kovetskiy) [inline src/network/networkd-link.c]: s/Invalied/Invalid/

**Comment by ssahani** [comment]
Thanks updated .

**Comment by georgmu** [comment]
shouldn't the value of accept_ra sysctl be set to "2" in case it is explicitly activated by the config and ip_forward is enabled? It wouldn't have any effect otherwise

See Documentation http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/Documentation/networking/ip-sysctl.txt?id=v4.2#n1308

It more ore less should be:
-1 -> write "1" (default)
0 -> write "0"
1 -> write "2"

**Comment by poettering** [comment]
ah, yuck. @georgmu you are right, but this should be filed as new issue.

**Comment by poettering** [comment]
Files a new issue in #1340 about this

**Pull request merged**